server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CCSUsageTelemetryIT.java

```diff
@@ -0,0 +1,466 @@
+
+        searchRequest = makeSearchRequest("*", REMOTE1 + ":" + remoteIndex);
+        assertResponse(cluster(LOCAL_CLUSTER).client(nodeName).search(searchRequest), Assert::assertNotNull);
+        telemetry = getTelemetrySnapshot(nodeName);
+        assertThat(telemetry.getTotalCount(), equalTo(2L));
+        assertThat(telemetry.getFeatureCounts().get(WILDCARD_FEATURE), equalTo(1L));
+
+        searchRequest = makeSearchRequest(localIndex, REMOTE2 + ":*");
+        assertResponse(cluster(LOCAL_CLUSTER).client(nodeName).search(searchRequest), Assert::assertNotNull);
+        telemetry = getTelemetrySnapshot(nodeName);
+        assertThat(telemetry.getTotalCount(), equalTo(3L));
+        assertThat(telemetry.getFeatureCounts().get(WILDCARD_FEATURE), equalTo(2L));
+
+        searchRequest = makeSearchRequest(localIndex, "*:" + remoteIndex);
+        assertResponse(cluster(LOCAL_CLUSTER).client(nodeName).search(searchRequest), Assert::assertNotNull);
+        telemetry = getTelemetrySnapshot(nodeName);
+        assertThat(telemetry.getTotalCount(), equalTo(4L));
+        assertThat(telemetry.getFeatureCounts().get(WILDCARD_FEATURE), equalTo(2L));
+    }
+
+    /**
+     * Test complete search failure
+     */
+    public void testFailedSearch() throws Exception {
+        Map<String, Object> testClusterInfo = setupClusters();
+        String localIndex = (String) testClusterInfo.get("local.index");
+        String remoteIndex = (String) testClusterInfo.get("remote.index");
+
+        SearchRequest searchRequest = makeSearchRequest(localIndex, "*:" + remoteIndex);
+        // shardId -1 means to throw the Exception on all shards, so should result in complete search failure
+        ThrowingQueryBuilder queryBuilder = new ThrowingQueryBuilder(randomLong(), new IllegalStateException("index corrupted"), -1);
+        searchRequest.source(new SearchSourceBuilder().query(queryBuilder).size(10));
+        searchRequest.allowPartialSearchResults(true);
+
+        String nodeName = cluster(LOCAL_CLUSTER).getRandomNodeName();
+        PlainActionFuture<SearchResponse> queryFuture = new PlainActionFuture<>();
+
+        // Borrowed from CrossClusterSearchIT#testSearchWithFailure
+        cluster(LOCAL_CLUSTER).client(nodeName).search(searchRequest, queryFuture);
+        assertBusy(() -> assertTrue(queryFuture.isDone()));
+
+        // We expect failure, but we don't care too much which failure it is in this test
+        ExecutionException ee = expectThrows(ExecutionException.class, queryFuture::get);
+        assertNotNull(ee.getCause());
+
+        CCSTelemetrySnapshot telemetry = getTelemetrySnapshot(nodeName);
+        assertThat(telemetry.getTotalCount(), equalTo(1L));
+        assertThat(telemetry.getSuccessCount(), equalTo(0L));
+        assertThat(telemetry.getTook().count(), equalTo(0L));
+        assertThat(telemetry.getTookMrtTrue().count(), equalTo(0L));
+        assertThat(telemetry.getTookMrtFalse().count(), equalTo(0L));
```

Comment:
I intend to deal with failure reasons on the next iteration. Generally, right now failure reasons are a bit broken because cancelled check seems to be too wide and probably much more fine grained checks need to be added. I'll add a TODO here.